Source/TutorialController.swift

```diff
@@ -30,96 +30,74 @@ public class TutorialController: PagesController {
         scrollView = subview as? UIScrollView
         scrollView?.delegate = self
       }
     }
-  }
-
-  override public func viewDidDisappear(animated: Bool) {
-    super.viewDidDisappear(animated)
 
-    NSNotificationCenter.defaultCenter().removeObserver(
-      self,
-      name: UIDeviceOrientationDidChangeNotification,
-      object: nil)
+    animateAtIndex(0, perform: { animation in
+      animation.play()
+    })
   }
 
-  override public func add(viewControllers: [UIViewController]) {
-    super.add(viewControllers)
-  }
+  // MARK: Public methods
 
   public override func goTo(index: Int) {
-    let pageCount = presentationCountForPageViewController(self)
-    let disableScroll = animationIndex != -1
+    super.goTo(index)
 
-    if index >= 0 && index < pageCount {
+    if index >= 0 && index < pagesCount {
       let reverse = index < animationIndex
       if !reverse {
         animationIndex = index
       }
 
-      if let animations = animationLayer[animationIndex] {
-        for animation in animations {
-          checkAnimation(animation)
-
-          if disableScroll {
-            scrollView?.delegate = nil
-          }
-
-          if reverse {
-            animation.playBack()
-          } else {
-            animation.play()
-          }
+      for slide in slides {
+        if reverse {
+          slide.goToLeft()
+        } else {
+          slide.goToRight()
         }
       }
-    }
 
-    super.goTo(index)
-  }
-
-  // MARK: device orientation
+      scrollView?.delegate = nil
 
-  func didRotate() {
-    if let animations = animationLayer[animationIndex] {
-      for animation in animations {
-        animation.rotate()
-      }
-    }
-    for backViewModel in backLayer {
-      backViewModel.rotate()
+      animateAtIndex(animationIndex, perform: { animation in
+        if reverse {
+          animation.playBack()
+        } else {
+          animation.play()
+        }
+      })
     }
   }
 }
 
-// MARK: - Back layer
+// MARK: Content
 
 extension TutorialController {
 
-  public func addBackViewModels(backViewModels: [BackViewModel]) {
-    for backViewModel in backViewModels {
-      addBackViewModel(backViewModel)
+  public override func add(viewControllers: [UIViewController]) {
+    for controller in viewControllers {
+      if controller is SlideController {
+        slides.append((controller as! SlideController))
+      }
     }
+    super.add(viewControllers)
   }
 
-  public func addBackViewModel(backViewModel: BackViewModel) {
-    backLayer.append(backViewModel)
-    view.addSubview(backViewModel.view)
-    view.sendSubviewToBack(backViewModel.view)
-    backViewModel.place()
+  public func addToScene(elements: [Content]) {
+    for content in elements {
+      scene.append(content)
+      view.addSubview(content.view)
+      view.sendSubviewToBack(content.view)
+      content.layout()
+    }
   }
 }
 
-// MARK: - Animation layer
+// MARK: Animations
```

Comment:
Might be better if we add the `-` to generate the line.

This:
`// MARK: - Animations`

Instead of this:
// MARK: Animations